src/main/scala/org/broadinstitute/dsde/workbench/sam/dataAccess/PostgresAccessPolicyDAO.scala

```diff
@@ -1686,13 +1690,86 @@ class PostgresAccessPolicyDAO(
         .toSet
     }
   }
-  override def filterResources(
-      samUser: SamUser,
+
+  private val publicResourcesCache: java.util.Map[ResourceTypeName, Seq[FilterResourcesResult]] =
+    Collections.synchronizedMap(new PassiveExpiringMap(1, TimeUnit.HOURS))
```

Comment:
1 hour seems like a while, am i correct in thinking that if someone changes a public resource itll take an hour to propagate that change? Do you see that causing any problems?

Reply:
I don't think it would be a problem. Publicizing workspaces is a manual process undertaken by Support, so we can just instruct them that it can take up to an hour for a public workspace to appear.